taskcluster/app_services_taskgraph/transforms/toolchain.py

```diff
@@ -4,16 +4,37 @@
 
 from __future__ import absolute_import, print_function, unicode_literals
 
+import subprocess
+
 from taskgraph.transforms.base import TransformSequence
 from taskgraph.util.schema import resolve_keyed_by
 
 transforms = TransformSequence()
 
+# TODO: Bug 1637695 to be removed once we retire these old indexes
+TOOLCHAIN_OLD_INDEX = {
+    'android': 'index.project.application-services.application-services.build.libs.android.{sha}',
+    'desktop-linux': 'index.project.application-services.application-services.build.libs.desktop.linux.{sha}',
+    'desktop-macos': 'index.project.application-services.application-services.build.libs.desktop.macos.{sha}',
+    'desktop-win32-x86-64': 'index.project.application-services.application-services.build.libs.desktop.win32-x86-64.{sha}',
+}
+
+
+def git_sha_for_directory(directory):
```

Comment:
Good point; I didn't bother with it initially since this is temporarily anyway but nevertheless, why not optimize this if we can, even if short-lived for a few weeks. I've added memoization 👍